_posts/2019-01-15-recursion.md

```diff
@@ -0,0 +1,313 @@
+---
+author: Sophie DeBenedetto
+author_link: https://github.com/sophiedebenedetto
+categories: recursion
+date: 2019-01-15
+layout: post
+title: Understanding Recursion with Elixir
+excerpt: >
+  De-mystify the concept of recursion and gain a deeper understanding of how and why to use it by writing our very own recursive function in Elixir.
+---
+
+"Recursion" can be a scary word for those of us unfamiliar with its applications. In this post, we'll de-mystify the concept of recursion and gain a deeper understanding of how and why to use it by writing our very own recursive function in Elixir.
+
+## What is Recursion
+
+In short, "recursion" is when a function calls itself. First we'll look at a contrived example. Later on in this post we'll build a more practical recursive function.
+
+Below we've defined a function `RecursionPractice.hello_world/0`, that calls itself:
+
+```elixir
+defmodule RecursionPractice do
+  def hello_world do
+    IO.puts "Hello, World!"
```

Comment:
For consistency with other functions I think we should use parenthesis for these `IO` module functions:

```suggestion
    IO.puts("Hello, World!")
```